### ISSUE TITLE
Creating products in different shops with the same name makes one unreachable

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2831]
### Expected behavior
Each product should be accessible by clicking on its link/image

### Actual Behavior
Clicking on either product brings you to the same product, one is inaccessible

### Steps to Reproduce the Behavior
1. Create a second shop and login as merchant admin
1. Create a product "Dumb Product" and publish
1. Go back to marketplace admin and create a product called "Dumb Product" and publish
1. Click on the second product you created
1. Observe that the first product is shown

(this might get fixed when shop prefixes are fixed? Or do we need to make handles unique across shops?)

### Versions
```
Node: 7.10.0
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.2-ce
```

**Comment by Akarshit** [comment]
Similar to #2560. I am working on a PR.

**Comment by spencern** [comment]
@Akarshit That's a great question as to why we're making JSON there and then parsing it, I don't have a good answer. @kieckhafer or @mikemurray  do you know who wrote that?

**Comment by spencern** [comment]
@zenweasel would you consider this to be a blocker for marketplace?

**Comment by kieckhafer** [comment]
@spencern not sure, git history says @aaronjudd but he could have just been the one to merge it in. This does not look familiar to me, pretty confident I didn't originally write it...

It looks like all it's doing is adding quotes around non-boolean values, perhaps that's required and it wasn't happening otherwise?

**Comment by zenweasel** [comment]
@spencern Yeah, I don't know how likely it will be to have products with the same name, but when you do it's a pretty bad bug.

**Comment by Akarshit** [comment]
No problem, I have left the old code as it is and written the new code to directly use object.

**Comment by spencern** [comment]
> do we need to make handles unique across shops?
This is a good question, it would be nice if each shop could have products with identical handles without causing problems, but we probably need to finish the #2760 first

**Comment by zenweasel** [comment]
@Akarshit Was this resolved by #2846?

**Comment by Akarshit** [comment]
@zenweasel Yes this can be closed.

**Comment by zenweasel** [comment]
Closed via #2846
